packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -24,7 +24,7 @@ export async function handleHotUpdate(
 	{ config, logging, compile }: HandleHotUpdateOptions
 ) {
 	let isStyleOnlyChange = false;
-	if (ctx.file.endsWith('.astro')) {
+	if (ctx.file.endsWith('.astro') && isCached(config, ctx.file)) {
```

Comment:
this is the only change. If the file is not cached then there's no reason to attempt this short-circuit path which will always cause false-positives.

Reply:
👍🏻 Makes total sense, nice catch!